tests/ProjectsMetadataTest.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ManageApiTest;
+
+use Keboola\ManageApi\Client;
+use Keboola\ManageApi\ClientException;
+use Keboola\ManageApi\ProjectRole;
+
+class ProjectsMetadataTest extends ClientTestCase
+{
+    public const TEST_METADATA = [
+        [
+            'key' => 'test_metadata_key1',
+            'value' => 'testval',
+        ],
+    ];
+
+    public const PROVIDER_USER = 'user';
+
+    private $organization;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->client->addUserToMaintainer($this->testMaintainerId, ['email' => 'spam+spam@keboola.com']);
+
+        foreach ($this->client->listMaintainerMembers($this->testMaintainerId) as $member) {
+            if ($member['id'] === $this->normalUser['id']) {
+                $this->client->removeUserFromMaintainer($this->testMaintainerId, $member['id']);
+            }
+
+            if ($member['id'] === $this->superAdmin['id']) {
+                $this->client->removeUserFromMaintainer($this->testMaintainerId, $member['id']);
+            }
+        }
+
+        $this->organization = $this->client->createOrganization($this->testMaintainerId, [
+            'name' => 'My org',
+        ]);
+
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUserWithMfa['email']]);
+        $this->client->removeUserFromOrganization($this->organization['id'], $this->superAdmin['id']);
+    }
+
+    public function testNormalUserCannotAddMetadata(): void
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->superAdmin['email']]);
+        $projectId = $this->createProjectWithSuperAdminMember($this->organization['id']);
+
+        try {
+            $this->normalUserClient->setProjectMetadata($projectId, self::PROVIDER_USER, self::TEST_METADATA);
+            $this->fail('Should fail.');
+        } catch (ClientException $e) {
+            $this->assertEquals(403, $e->getCode());
+        }
+    }
+
+    public function testSuperAdminCanAddMetadataInOrgWithAllowAutoJoinFalse()
```

Comment:
Tohle je takové na hraně mi přijde. Ale asi je to na úrovni přidávání uživatelů - na to taky není potřeba být v projektu IMHO